crates/core/src/models/balance.rs

```diff
@@ -13,3 +17,25 @@ pub struct TokenBalances {
     pub address: Address,
     pub token_balances: Vec<TokenBalance>,
 }
+
+#[pin_project::pin_project]
```

Comment:
what is this?

Reply:
I was just in the process of leaving a comment on this, it basically allows you to avoid having to write unsafe code to `pin` your future. Here is the [blog post](https://blog.cloudflare.com/pin-and-unpin-in-rust/) I was reading to implement the PR.